### DISCUSSION TITLE
Climate entity setup for Ariston ebusd

### DISCUSSION THREAD

**Discussion by michalmie** [wrongisthenewright/ebusd-configuration-ariston-bridgenet#7] [Q&A] [answered]
Hi

First, I want to express my appreciation for the incredible work You've done. It's truly amazing to see what can be achieved with platform and community behind it.

CC: @wrongisthenewright 

I've been attempting to configure a climate entity to appear as a thermostat on my Home Assistant dashboard, but regrettably, I haven't achieved the desired outcome. Could You please share some insights how to achieve it?

![image](https://github.com/wrongisthenewright/ebusd-configuration-ariston-bridgenet/assets/6077973/593b1db9-24a2-4edc-8baa-f7becfb79b3d)

**Comment by wrongisthenewright** [comment]
Hi
I'm sorry but cannot help on this, I'm not using the climate entity in HA as I'm using it mainly for monitoring the system and for some really basic automation (start cooling on PV surplus during summer etc). I'm not trusting my skills with HA for something more serious, I leave the general running of the system to the Ariston Sensys.

In theory it should be possible to use the MQTT climate entity with crafted messages for various settings. 
Ebusd mqtt integration use xxxxx/set topic to write the values.

Hope it helps.

**Reply by wrongisthenewright** [reply, marked as answer]
I've looked into this and I think it's not possible to use the native MQTT climate entity with my Bridgenet protocol connected via Ebusd, I find at least 2 problems:

1.  the climate integration does not support the "Summer mode" of the appliance (no HVAC but DHW active)
2.  the various modes are activated via different MQTT topics so it's not possible to use the "mode_state_topic" and "mode_command_topic" confiuration variables since these support only 1 MQTT topic each.

You can maybe use some templating to render the various MQTT topics into "modes" as the MQTT HVAC integration needs, but I can't help there as I've almost no skill on jinja2 coding.

I've made a custom dashboard to command the HP, it this I put all the most commonly used entity that have writing enabled: from there I can set the temps, the modes (but not the scheduling) and so on. For the mode change I've made an automation that stops cooling if heating is enabled and vice versa, maybe it's redundant but I played it safely...

**Reply by michalmie** [reply]
Thank you for your dedication and the time you've spent. If I manage to find a solution, I will definitely share it here.

**Reply by MBRetrofit** [reply]
> Thank you for your dedication and the time you've spent. If I manage to find a solution, I will definitely share it here.

All good?

**Comment by wrongisthenewright** [comment]
Hi,
I haven't tested it because I'm lazy and because I leave the running of my systtem to the Ariston thermostat but it seems this custom thermostat:

https://github.com/swingerman/ha-dual-smart-thermostat

supports heating and cooling via dedicated switches. Given that my CSV with the _templates.csv file exposes switches to HA it should be possible to adapt it to be used with Ariston devices.

I think heater and cooler should be set to the switches created by autodiscovery. target_sensor should point to Zx (z1, z2...) DAY temp, night temp should be set to minimum as the thermostat will modify the day temp to the desired level 


What you'll loose:
- ariston built-in scheduler (as night temp will never be used/useful)
- DHW sould be enabled/disabled via HA switch or via Ariston thermostat

Beware that non volatile parameters (as temperature configurations) are written on the non volatile memory of the various PCBs, since these are "normal" flash chips not suited for continuous write operations if you keep to modify them frequently the life of your PCBs will undoubtedly be compromised... I think it is worth only for those that are on a boiler only configurtiton without a functioning room thermostat like the Sensys
